lib/rag/generation.ex

```diff
@@ -18,7 +18,7 @@ defmodule Rag.Generation do
           context: String.t() | nil,
           context_sources: list(String.t()),
           prompt: String.t() | nil,
-          response: String.t() | nil,
+          response: String.t() | Enumerable.t() | nil,
```

Comment:
nice :+1: 

We could make it more readable by introducing a `response()` type above, like
```elixir
@type response :: String.t() | Enumerable.t()
```
and then use that type in the typespecs in this file wherever there is a response.
```suggestion
          response: response() | nil,
```
What do you think?

Reply:
From a design perspective, which would be better

```
@type response :: String.t() | Enumerable.t()
```

or 

```
@type response :: String.t() | Enumerable.t() | nil
```

Reply:
I would say the first. The response can be a String or an enumerable.

If it's nil, it means there is no response (yet).

Reply:
Of course, that makes sense.